graphene_pydantic/inputobjecttype.py

```diff
@@ -0,0 +1,146 @@
+    """
+    Construct all the fields for a PydanticInputObjectType.
+
+    NOTE: Currently simply fetches all the attributes from the Pydantic model
+    `__fields__`. In the future we hope to implement field-level overrides that
+    we'll have to merge in.
+    """
+    excluded: T.Tuple[str, ...] = ()
+    if exclude_fields:
+        excluded = exclude_fields
+    elif only_fields:
+        excluded = tuple(k for k in model.__fields__ if k not in only_fields)
+
+    fields_to_convert = (
+        (k, v) for k, v in model.__fields__.items() if k not in excluded
+    )
+
+    fields = {}
+    for name, field in fields_to_convert:
+        converted = convert_pydantic_input_field(
+            field, registry, parent_type=obj_type, model=model
+        )
+        registry.register_object_field(obj_type, name, field)
+        fields[name] = converted
+    return fields
+
+
+# TODO: implement an OverrideField of some kind
+
+
+class PydanticInputObjectType(graphene.InputObjectType):
+    """Graphene InputObjectType that knows how to map itself to a Pydantic model defined in its nested `Meta` class."""
+
+    @classmethod
+    def __init_subclass_with_meta__(
+        cls,
+        model: type = None,
+        registry: Registry = None,
+        skip_registry: bool = False,
+        only_fields: T.Tuple[str, ...] = (),
+        exclude_fields: T.Tuple[str, ...] = (),
+        id=None,
+        _meta=None,
+        **options,
+    ):
+        assert model and issubclass(
+            model, pydantic.BaseModel
+        ), f'You need to pass a valid Pydantic model in {cls.__name__}.Meta, received "{model}"'
+
+        assert isinstance(
+            registry, (Registry, None.__class__)
+        ), f'The attribute registry in {cls.__name__} needs to be an instance of Registry, received "{registry}".'
+
+        if only_fields and exclude_fields:
+            raise ValueError(
+                "The options 'only_fields' and 'exclude_fields' cannot be both set on the same type."
+            )
+
+        if not registry:
+            registry = get_global_registry(PydanticInputObjectType)
+
+        pydantic_fields = yank_fields_from_attrs(
+            construct_fields(
+                obj_type=cls,
+                model=model,
+                registry=registry,
+                only_fields=only_fields,
+                exclude_fields=exclude_fields,
+            ),
+            _as=graphene.InputField,
+            sort=False,
+        )
+
+        if not _meta:
+            _meta = PydanticInputObjectTypeOptions(cls)
+
+        _meta.model = model
+        _meta.registry = registry
+
+        if _meta.fields:
+            _meta.fields.update(pydantic_fields)
+        else:
+            _meta.fields = pydantic_fields
+
+        _meta.id = id or "id"
+
+        super().__init_subclass_with_meta__(_meta=_meta, **options)
+
+        if not skip_registry:
+            registry.register(cls)
+
+    @classmethod
+    def resolve_placeholders(cls):
```

Comment:
Nit: this looks at first glance to be the same as `PydanticObjectType.resolve_placeholders`, it would be good not to duplicate this code.